### PR TITLE
feat(date range): DatePicker with range

### PR DESCRIPTION
**Pull request by yyanwang**
**What is the problem this PR is trying to solve?**
Date range:
http://guidelines.talend.com/document/92132#/ui-controls/date-time-picker-1561102408
This PR is only to extend current DatePicker to be able to show date range. other changes will come in following PRs.
**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by yyanwang** [comment]
Story:

**Comment by yyanwang** [comment]
Hi @jmfrancois ,
thanks for review! 
yes, I am aware that there should be 2 input fields, to select startDate/endDate. :wink:
this PR contains only changes to DatePicker, which is a stateless component, to be able to show range. state and behavior will be handled in a `DateRang.Manager` - a context provider.
so the structure of Single date picker is like:
```
<Date.Manager>
    <DatePicker />
</Date.Manager>
```
and for Range picker, it's like:
```
<DateRange.Manager>
    <DatePicker from />
    <DatePicker to />
</DateRagne.Manager>
```
Single date picker and Range Date picker work under different Manager which will handle the state and behavior.
A new DateRange.Manager will be added on my next PR(I was trying to make smaller PR so it will be easier for review).

**Comment by yyanwang** [comment]
Hi @jmfrancois 
Thanks for review! 
I noticed that there should be 2 inputs for range picker. :wink: 

This PR contains just changes to `DatePicker` which is a controlled component, to make it be able to show range. and this `DatePicker` will be reused for single date picker and range date picker.

The state and behavior will be handled in a `DateRange.Manager`, it's a context provider. for single date picker will use `Date.Manager`, so the logic will not be mixed together.

The problem you see that single date and range can show at the same time is because `DateRange.Manager` is not ready yet, it will come in next PR.(I try to make smaller PRs so it will be easier to review).

I hope I made it clear 😅 , if I am not maybe @jsomsanith can help make some comments.

File: packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js [hunk 1]
```diff
@@ -112,9 +126,15 @@ class DatePicker extends React.PureComponent {
 									(selectedInCurrentCalendar && selected) ||
 									(!selectedInCurrentCalendar && day === 1);
 
+								const isStart = isSameDay(date, startDate);
+								const isEnd = isSameDay(date, endDate);
+								const isMiddle = isWithinRange(date, addDays(startDate, 1), subDays(endDate, 1));
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js]:
```suggestion
								const isMiddle = !isStart  && !isEnd  && isWithinRange(date, startDate, endDate);
```

File: packages/components/src/DateTimePickers/pickers/CalendarPicker/CalendarPicker.component.js [hunk 2]
```diff
@@ -27,6 +36,8 @@ class CalendarPicker extends React.Component {
 				year: getYear(initialCalendarDate),
 			},
 			selectedDate,
+			startDate,
+			endDate,
```
Comment (jsomsanith-tlnd) [inline packages/components/src/DateTimePickers/pickers/CalendarPicker/CalendarPicker.component.js]: Do not set them in state, just pass the props in the jsx. Here you duplicate the value, and never update it. And it's only to passe them in the jsx, just pass the props

**Comment by yyanwang** [comment]
story link:
http://2491.talend.surge.sh/components/?path=/story/datepicker--date-range-picker-no-input

**Pull request merged**